Source/Core/VideoCommon/TextureDecoder_Common.cpp

```diff
@@ -374,37 +374,37 @@ void TexDecoder_Decode(u8* dst, const u8* src, int width, int height, int texfor
 
 static inline u32 DecodePixel_IA8(u16 val)
 {
-  int a = val & 0xFF;
-  int i = val >> 8;
-  return i | (i << 8) | (i << 16) | (a << 24);
+  u8 a = val & 0xFF;
```

Comment:
These are actually truncating without an explicit cast and can lead to warnings. The result of these operations in this context is an `int`, Ditto for similar occurrences below.